### ISSUE TITLE
Segmentation fault on x86_64 with MacOS Virtualization Framework enabled

### ISSUE THREAD

**Issue by antoniofrignani** [docker/for-mac#6824]
- [X] I have tried with the latest version of Docker Desktop
  - [X] I have tried disabling enabled experimental features
  - [X] I have uploaded Diagnostics
  - Diagnostics ID: 42EDF249-C0A8-4A3B-A709-4B2ADD41E5EF/20230505143852

### Expected behavior

After a fresh local install of a Laravel project + Breeze (Vue), trying to build for production (`sail npm run build`) or starting a local hmr server (`sail npm run dev`) should work properly (using Docker for Mac on Intel Chip, v4.19.0)

### Actual behavior

Both commands thows a Segmentation Fault error without further info.

```
  VITE v4.3.3  ready in 727 ms

  ➜  Local:   http://localhost:5173/
  ➜  Network: http://172.240.7.5:5173/
  ➜  press h to show help

  LARAVEL v10.9.0  plugin v0.7.4

  ➜  APP_URL: http://mysite.test
Segmentation fault
```

### Information

Switching back to Docker for Mac v4.18.0 works as expected.

Possible workaround: disable Virtualization framework on v4.19+

### Steps to reproduce the behavior

  1.  Create a fresh Laravel project (`curl -s "https://laravel.build/example-app" | bash`)
  2. Install Laravel Breeze package (https://laravel.com/docs/10.x/starter-kits#laravel-breeze-installation) with the Vue stack
  3. Run `sail npm run dev` should throw the Segmentation Fault error

**Comment by MihaelaStoica** [comment]
@antoniofrignani, would it be possible to submit a [diagnostic report](https://docs.docker.com/desktop/troubleshoot/overview/#diagnose) and then add the Diagnostics ID here?

**Comment by antoniofrignani** [comment]
@MihaelaStoica i've just updated my comment with a diagnostic ID

**Comment by irobin591** [comment]
I also have this issue.

There is no need to create a full Laravel project though. Installing Vite is enough to reproduce this issue:
- Create a new Vite project via `docker run -it --rm -v "$PWD":/usr/src/app -w /usr/src/app node npm create vite@latest`
- Change directory into the new Vite project: `cd vite-project`
- Install the Vite package via npm: `docker run -it --rm -v "$PWD":/usr/src/app -w /usr/src/app node npm install`
- Build for production or start a local hmr server: `docker run -it --rm -v "$PWD":/usr/src/app -w /usr/src/app node npm run dev` / `docker run -it --rm -v "$PWD":/usr/src/app -w /usr/src/app node npm run build`

Downgrading to 4.18.0 via https://docs.docker.com/desktop/release-notes/#4180 without changing the version of Vite is also a working workaround for me.

I also have access to a MacBook of a different model where this issue does not occur, so this might be hard to reproduce. 

Specs of the MacBook that has the issue:
```
13" MacBook Pro 2020
CPU: 2,3GHz Quad-Core Intel Core i7
RAM: 16 GB 3733 MHz LPDDR4X 
OS: Ventura 13.3.1 (a)
```

Specs of the MacBook without the issue:
```
15" MacBook Pro 2017
CPU: 2,9GHz Quad-Core Intel Core i7
RAM: 16 GB 2133 MHz LPDDR3
OS: Ventura 13.3.1 (a)
```

**Comment by tisba** [comment]
I reported this issue to the yarn project (in hindsight not sure if I should have filed it here or at the node project instead). This issue here looks very similar to what I reported at https://github.com/yarnpkg/berry/issues/5420. The issue at the yarn repository contains a minimal case to reproduce.

For me, this issue happens 100% of the time, for coworkers (almost identical hardware) it does not fail at all.

**Comment by antoniofrignani** [comment]
@irobin591 yes the issue is not direcly related to the framework itself, thanks for reproducing it with bare minimum boilerplate.

I'll watch the issue on the node/yarn side too since this seems strictly related, thanks @tisba for reporting it here.

**Comment by tisba** [comment]
I can confirm that downgrading to Docker Desktop 4.18.0 (104112) fixes the issue for `yarn install` seg fauting (see [comment](https://github.com/docker/for-mac/issues/6824#issuecomment-1538881539) above). Thanks for the hint @irobin591 - I can at least build assets again :)

**Comment by tisba** [comment]
@sam-thibault @MihaelaStoica: Can we provide something else to help with acknowledging/analysing the issue? (looking at the `status/0-more-info-needed` label)

**Comment by julienallexandre** [comment]
I have the same problem on a Laravel project with sail when I do the command `sail npm run dev` I get : 

> dev
> vite

> Segmentation fault

Docker version 23.0.5, build bc4487a
Docker Compose version v2.17.3
Node v18.16.0
NPM 9.6.7

**Comment by julienallexandre** [comment]
@tisba thanks yes that's what I did :)

**Comment by doyeka** [comment]
Ran into the same issue today. I was using:
Node v19
Yarn 3.5.0
2.3 GHz Quad-Core Intel Core i7
Docker 4.19.0 & 4.20.1

I continually ran into silently failing yarn installs. Eventually, I tried a yarn install in the Docker terminal itself and received a seg fault.

Shoutout to you @tisba - I was losing my mind all afternoon.

Please fix this issue. @docker-robot

**Comment by tisba** [comment]
I can confirm that with Docker Desktop 4.20.1 (110738) the problem persists under x86 (macOS 13.3.1 (a) (22E772610a)).

None of my Apple Silicon devices show this problem, also tested with latest Docker Desktop 4.20.1.

**Comment by sergioharo** [comment]
I have some problem on i9 intel iMac. I can reproduce on latest docker desktop. Downgrading to 4.18 fixes issue but has networking problems of its own on Ventura 13.4.

Interestingly, I can reproduce with node v18 and 20 buster/bullseye images. However node v16 works fine.

**Comment by tisba** [comment]
Updated to macOS 13.4 (22F66) and the problem still persists.

**Comment by charlon** [comment]
As an added piece of information... I have 2 development setups I am currently working in and experiencing this same issue. I am using the bullseye images for my containers.

- Home: i9 Intel iMac, node versions v14, v16, v17 can build successfully, v18-20 results in seg fault
- Work: i5 Intel iMac, node versions v14-20 can all build successfully

Both machines are running:  macOS 13.4 / Docker Desktop v4.20.1 / Node v18.16.0

**Comment by antoniofrignani** [comment]
Same problem with version 4.20.1

**Comment by taro-ishihara** [comment]
I had same issue.

MacBook Pro (13-inch, 2020, Four Thunderbolt 3 ports)
Monterey 12.6.2 Intel Core i7

But it's fixed after downgraded to 4.18.0 without any changes of dockerfile, thank you.

**Comment by RogierW** [comment]
I had the same issue, but I use Laravel Mix instead of Vite. Downgrading Docker desktop to 4.18.0 did also the trick for me.

**Comment by tisba** [comment]
Still preproducable with 4.21.0 (113844) on Ventura 13.4.1 (22F82). Do the issue labels need to be updated to reflect that the bug is still present, @sam-thibault?

**Comment by Tynael** [comment]
Managed to reproduce the issue on macOS Ventura 13.4.1 (22F82) with Docker Desktop 4.20.1 (110738) and Docker version 24.0.2, build cb74dfc

**Comment by antoniofrignani** [comment]
Problem still relevant with latest 4.21.1 version

**Comment by KFoxder** [comment]
@antoniofrignani confirming as well the issue is still relevant on `4.21.1` with an Intel Mac.

**Comment by sotatek-manhnguyen8** [comment]
any update for Docker 4.21.1 with Intel mac?

**Comment by MajesticGray** [comment]
> any update for Docker 4.21.1 with Intel mac?

yarn still crashing on 4.21.1

**Comment by pablo-psi** [comment]
Same issue on this laptop, still failing:
```
13" MacBook Pro 2020
CPU: 2GHz Quad-Core Intel Core i5
RAM: 16 GB 3733 MHz LPDDR4X 
OS: Ventura 13.2.1
```

Both Vite and Vitest were returning `Segmentation fault`. I tried everything I could think of and the only workaround I found was downgrading Docker from v4.21.1 to v4.18.0 - Thank you guys so much!

**Comment by 8de2fdb0** [comment]
@sam-thibault is this somehow on the radar.  I am following this ticket for quiet some time and just like anyone here I am stuck on the last working release.

I suppose nobody found the root cause yet but might be easier for a docker dev to debug where that segfault comes from.

**Comment by AbbyRH** [comment]
I've had a similar issue with running the AWS cdk inside containers under Docker Desktop on Mac. After attempting the version downgrade without success. I tried un-checking "Use Virtualization framework" and this appears to have resolved my issues. 
It may be worth trying here since AWS cdk is a nodejs package there could be a common thread there.

**Comment by antoniofrignani** [comment]
Confirmed that the problem is somewhere in-between Docker Engine and MacOS Virtualization Framework, it now works disabling the latter on v 4.22.

This seems to be the only working workaround as of now. Thanks @AbbyRH for sharing your solution.

**Comment by tisba** [comment]
@antoniofrignani why did you close this issue? I'd prefer to not having to open a new issue myself, tbh. Maybe you can re-open and adjust the title/description accordingly? Thanks.

**Comment by gulden-tv** [comment]
I confirm that switching off Virtualization Framework fixes for me too issue with yarn seg fault
```
MacBook Air Retina, 13-inch, 2020
1,1 GHz Quad-Core Intel Core i5
Docker 4.21.1
```

**Comment by Felixclement** [comment]
I can confirm that this issue does not happen on mac ARM, my collegue with intell mac same project as the issue

**Comment by denisvmedia** [comment]
I can confirm it on

MacBook Model: MacBookAir9,1
Processor Information: Intel(R) Core(TM) i7-1060NG7 CPU @ 1.20GHz
Memory (RAM) Information: 16 GB
macOS Version: Ventura 13.5.1
Docker Version: 4.22.1 (118664) / Engine 24.0.5

**Comment by NickSpaghetti** [comment]
I came across this thread this morning, and turning off MacOS Virtualization Framework solves the issue.

MacBook Model: MacBook Pro 13-inch, 2020, Four Thunderbolt 3 ports
Processor Information: Intel(R) Core(TM) i7-1068NG7 CPU @ 2.30GHz
Memory (RAM) Information: 32 GB
macOS Version: Ventura 13.5.1
Docker Version: 4.22.1 (118664) / Engine 24.0.5

**Comment by MajesticGray** [comment]
Still happening on 4.23.0, which automatically enables virtualization again

**Comment by taniguchi15** [comment]
v4.23.0 The setting location had been moved.
Settings >  General
Change `VirtioFS` to `osxfs(Legacy)` and disable `Use Virtualization framework`.

**Comment by tisba** [comment]
Happy to report, that for macOS 14.0 (Sonoma) and Docker Desktop 4.24.0 (122432) I can no longer reproduce the issue. I have not checked all Docker Desktop versions since my [last comment](https://github.com/docker/for-mac/issues/6824#issuecomment-1618243886), so I don't know if this was fixed by Docker or Apple 🤷 

I used https://github.com/docker/for-mac/issues/6824#issuecomment-1544374262 in an attempt to reproduce as well as a private code base I'm working on.

**Comment by tisba** [comment]
Do the issue labels need to be updated to reflect that the bug is still present, @bsousaa, @sam-thibault? Also: Since it seems so easily reproducible by others, is there anything the Docker team can use to fix the underlying issue? 🙏

**Comment by treckstar** [comment]
Definitely still happening, just downgraded to v4.21, v4.19, and v4.18 and docker image finally built. Using `pnpm` on 2020 Intel iMac Ventura 13.6, while trying to `pnpm install` mainly. Specifically was trying to build the Vuero template Dockerfile.
> Segmentation Fault error without further info.

**Comment by bsousaa** [comment]
@tisba @treckstar can you test with this build? https://desktop-stage.docker.com/mac/main/arm64/123434/Docker.dmg

**Comment by bsousaa** [comment]
@tisba https://desktop-stage.docker.com/mac/main/amd64/123434/Docker.dmg

**Comment by tisba** [comment]
> @tisba [desktop-stage.docker.com/mac/main/amd64/123434/Docker.dmg](https://desktop-stage.docker.com/mac/main/amd64/123434/Docker.dmg)

I just ran the build (`4.25.0 (123434)`) with https://github.com/docker/for-mac/issues/6824#issuecomment-1544374262. And it continues to segfault.

<details>
<summary>Output of test script</summary>

```
removed 1 package, and changed 50 packages in 6s

18 packages are looking for funding
  run `npm fund` for details
uname -a: Linux 371342092a98 6.4.16-linuxkit #1 SMP PREEMPT_DYNAMIC Wed Oct  4 10:01:08 UTC 2023 x86_64 GNU/Linux
Node version: v18.16.1
Yarn version: 3.5.1
npm version: 9.6.5


Running: yarn install
➤ YN0065: Yarn will periodically gather anonymous telemetry: https://yarnpkg.com/advanced/telemetry
➤ YN0065: Run yarn config set --home enableTelemetry 0 to disable

➤ YN0000: ┌ Resolution step
➤ YN0000: └ Completed in 0s 329ms
➤ YN0000: ┌ Fetch step
➤ YN0013: │ @rails/ujs@npm:7.0.4 can't be found in the cache and will be fetched from the remote registry
➤ YN0000: ⠴ ========================================----------------------------------------
Segmentation fault
```

</details>



<details>
<summary>docker info</summary>

```
$ docker info
Client:
 Version:    24.0.6
 Context:    desktop-linux
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.11.2-desktop.5
    Path:     /Users/sebastian/.docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.22.0-desktop.2
    Path:     /Users/sebastian/.docker/cli-plugins/docker-compose
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.0
    Path:     /Users/sebastian/.docker/cli-plugins/docker-dev
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.20
    Path:     /Users/sebastian/.docker/cli-plugins/docker-extension
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v0.1.0-beta.8
    Path:     /Users/sebastian/.docker/cli-plugins/docker-init
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     /Users/sebastian/.docker/cli-plugins/docker-sbom
  scan: Docker Scan (Docker Inc.)
    Version:  v0.26.0
    Path:     /Users/sebastian/.docker/cli-plugins/docker-scan
  scout: Docker Scout (Docker Inc.)
    Version:  v1.0.7
    Path:     /Users/sebastian/.docker/cli-plugins/docker-scout

Server:
 Containers: 5
  Running: 3
  Paused: 0
  Stopped: 2
 Images: 76
 Server Version: 24.0.6
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 8165feabfdfe38c65b599c4993d227328c231fca
 runc version: v1.1.8-0-g82f18fe
 init version: de40ad0
 Security Options:
  seccomp
   Profile: unconfined
  cgroupns
 Kernel Version: 6.4.16-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 8
 Total Memory: 7.766GiB
 Name: docker-desktop
 ID: 2cbec23b-0661-4b62-9195-6bab049044a4
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: daemon is not using the default seccomp profile
```

</details>

**Comment by bsousaa** [comment]
@tisba can you share a diagnostics id please?

**Comment by excid3** [comment]
Ran into this today. Tested with 4.24.0 and 4.25.0 on Intel iMac.

Doesn't seem like Docker will give me a diagnostics ID?
<img width="492" alt="Screenshot 2023-10-11 at 4 27 57 PM" src="https://github.com/docker/for-mac/assets/67093/084112b2-2a34-408d-940d-813c29cecb09">

```bash
± /Applications/Docker.app/Contents/MacOS/com.docker.diagnose check

Starting diagnostics

[PASS] DD0027: is there available disk space on the host?
[PASS] DD0028: is there available VM disk space?
[PASS] DD0018: does the host support virtualization?
[PASS] DD0001: is the application running?
[FAIL] DD0017: can a VM be started? vm has not started: vm has not started
[FAIL] DD0016: is the LinuxKit VM running? prereq failed: can a VM be started?
[FAIL] DD0011: are the LinuxKit services running? prereq failed: is the LinuxKit VM running?
[FAIL] DD0004: is the Docker engine running? prereq failed: are the LinuxKit services running?
[PASS] DD0015: are the binary symlinks installed?
[FAIL] DD0031: does the Docker API work? prereq failed: is the Docker engine running?
[PASS] DD0013: is the $PATH ok?
[FAIL] DD0003: is the Docker CLI working? prereq failed: is the Docker engine running?
[FAIL] DD0038: is the connection to Docker working? prereq failed: is the Docker engine running?
[FAIL] DD0014: are the backend processes running? prereq failed: is the LinuxKit VM running?
[FAIL] DD0007: is the backend responding? prereq failed: are the backend processes running?
[PASS] DD0008: is the native API responding?
[FAIL] DD0009: is the vpnkit API responding? prereq failed: are the backend processes running?
[FAIL] DD0010: is the Docker API proxy responding? prereq failed: are the backend processes running?
[FAIL] DD0030: is the image access management authorized? prereq failed: is the Docker engine running?
[PASS] DD0033: does the host have Internet access?
[PASS] DD0018: does the host support virtualization?
[PASS] DD0001: is the application running?
[WARN] DD0017: can a VM be started? vm has not started: vm has not started
[WARN] DD0016: is the LinuxKit VM running? prereq failed: can a VM be started?
[WARN] DD0011: are the LinuxKit services running? prereq failed: is the LinuxKit VM running?
[WARN] DD0004: is the Docker engine running? prereq failed: are the LinuxKit services running?
[PASS] DD0015: are the binary symlinks installed?
[WARN] DD0031: does the Docker API work? prereq failed: is the Docker engine running?
[WARN] DD0032: do Docker networks overlap with host IPs? prereq failed: does the Docker API work?

Please note the following 6 warnings:

1 : The check: can a VM be started?
    Produced the following warning: vm has not started: vm has not started

The Docker engine runs inside a Linux VM. Therefore we must be able to start Virtual Machines.

2 : The check: is the LinuxKit VM running?
    Produced the following warning: prereq failed: can a VM be started?

The Docker engine runs inside a Linux VM. Therefore the VM must be running.

3 : The check: are the LinuxKit services running?
    Produced the following warning: prereq failed: is the LinuxKit VM running?

The Docker engine runs inside a Linux VM as a service. Therefore the services must have started.

4 : The check: is the Docker engine running?
    Produced the following warning: prereq failed: are the LinuxKit services running?

The Docker engine manages all containers and images on the host. Check the dockerd.log to see why it failed to start.

5 : The check: does the Docker API work?
    Produced the following warning: prereq failed: is the Docker engine running?

If the Docker API is not available from the host then Docker Desktop will not work correctly.

6 : The check: do Docker networks overlap with host IPs?
    Produced the following warning: prereq failed: does the Docker API work?

If the subnet used by a Docker network overlaps with an IP used by the host, then containers
won't be able to contact the overlapping IP addresses.

Try configuring the IP address range used by networks: in your docker-compose.yml.
See https://docs.docker.com/compose/compose-file/compose-file-v2/#ipv4_address-ipv6_address


Please investigate the following 1 issue:

1 : The test: can a VM be started?
    Failed with: vm has not started: vm has not started

The Docker engine runs inside a Linux VM. Therefore we must be able to start Virtual Machines.
```

**Comment by pablo** [comment]
Still an issue with 4.23.0 (120376) version. 

Upgraded to 4.24.2 (124339). Issue persists. 

Disabled MacOS Virtualization Framework: works.

**Comment by vpmedia** [comment]
Same issue here with Node v18 and v20, Node v16 works (no seg fault).
Only happening with my Mac with Intel, Mac with ARM works.

**Comment by lnovitz** [comment]
Downgrading Downgrading Docker desktop to 4.18.0 also worked for me (https://docs.docker.com/desktop/release-notes/#4180). I have a Macbook Pro 2020, 2 GHz Quad-Core Intel Core i5.

**Comment by alkami-jdalrymple** [comment]
Just upgraded to v4.25.0 and this appears to be an issue.

**Comment by markdegrootnl** [comment]
Docker Desktop 4.25.1 with NodeJS version 21 seems to be working.

**Comment by atli-c** [comment]
> Docker Desktop 4.25.1 with NodeJS version 21 seems to be working.

Can confirm v21 (21.1.0) seems to work without segmentation fault. Versions v18.18.2 and v20.8.0 crash with segfault.  
Docker Desktop 4.25.1 (128006) with (Virtualization framework + VirtioFS) on macOS Sonoma 14.1.1. Intel x86_64 CPU.

**Comment by MajesticGray** [comment]
> Can confirm that Docker Desktop 4.25.1 **still crashes** with NodeJS LTS (v20.9.0). On macOS 14.1.1 (23B81).

I can confirm it's crashing too

**Comment by carterwward** [comment]
Can confirm docker desktop 4.25.2 still crashes with npm version 9.8.0 on macOS 12.5 with Intel i5.

**Comment by hokkaido** [comment]
I can also confirm that I get segmentation faults with macOS 12.7.1, Docker Desktop 4.24.2 and Node:20.10.0 images. Changing to gRPC FUSE with Virtualization Framework off seems to be a workaround for me as well.

**Comment by web-engineer** [comment]
I had a very similar probelm with build failing with segfault. --no-cache did help in one project (think that was due to bad cached layers) but in a project today this was being super stubborn and refused to build for x86.

I've found a workaround that may help some of you - I had a hunch segfault was probably caused by lack of resources during the complilation process - 

```RUN docker-php-ext-install mysqli pdo_mysql gd zip```

This was the line in the Dockerfile on which I ended up getting the segfault - the fix was simply to break this up a bit -

```
RUN docker-php-ext-install mysqli pdo_mysql
RUN docker-php-ext-install gd
RUN docker-php-ext-install zip
```

Although this results in 3 layers it did complete the build, the builds always worked for arm, but I was getting segfaults trying to build images until I split this out.

This was in relation to getting a Lumen api docker build which is now running. There is definately something squiffy here though as ideally we want images with few layers and the package dependancies should have worked as a one line...

**Comment by dossy** [comment]
As an additional datapoint, I ran into what I imagine is this issue today with node v18.16.0 and running webpack 5.89.0, segfaulting on Docker 4.26.1 on Intel iMac 27" 2020 on macOS Ventura 13.3.1.

Switching from VirtioFS to gRPC FUSE and unchecking "Use Virtualization framework" and restarting Docker appears to have fixed the issue for me, as others have reported.

**Comment by krsyoung** [comment]
I think we're facing issues around the same area. For us, the seg fault shows when trying to install puppeteer (`node:20-slim` base) when Virtualization Framework / VirtioFS is enabled.

```
yarn install v1.22.19
...
error /app/node_modules/puppeteer: Command failed.
Exit code: 139
Command: node install.mjs
Arguments: 
Directory: /app/node_modules/puppeteer
Output:
Segmentation fault
```

This is with 2 GHz Quad-Core Intel Core i5, Docker for Mac 4.26.1 with Sonoma 14.2.1. I can find zero helpful logs / debug.

We can work around the issue by disabling Virtualization Framework and using gRPC FUSE.

I tried flipping switches for the better part of the day (disable resource saver, disable extensions etc..) to no avail.

Interestingly, moving to `node:21-slim` avoids the seg fault on `yarn install`. However, chrome headless then proceeds to hang. Chrome will work as expected if we disable Virtualization Framework and switching to gRPC FUSE.

So, there is a workaround but it does feel very slow!

**Comment by misstricky** [comment]
I ran into the same issue this morning on most recent docker, trying to run `npm run build` - downgrading to docker 4.18 fixed it for me

**Comment by schwemmer** [comment]
I can confirm that it is still happening with Docker Desktop Version 4.27.0 and Docker engine 25.0.1 on an Intel Mac with MacOS Ventura 13.2 when running `npm run dev`. Unchecking "Use Virtualization Framework" and using "gRPC FUSE" as file sharing implementation fixes it.

**Comment by unlikelyzero** [comment]
This is like an emotional support group for folks who lost half a day or more on debugging a problem that only a linux greybeard can solve

**Comment by JOSEPHGILBY** [comment]
My data point is that I have grpc FUSE checked AND I have 'use virtualization framework' checked and am no longer seeing the issue with vite and Laravel on an Intel Mac.

**Comment by MajesticGray** [comment]
Several months passed and still getting segfault whenever I enable virtualization framework, whether I choose FUSE or VirtioFS.

Docker  4.27.1, MacOS 14.2.1, Intel Core i5

**Comment by msassmann** [comment]
> Several months passed and still getting segfault whenever I enable virtualization framework, whether I choose FUSE or VirtioFS.
> 
> Docker 4.27.1, MacOS 14.2.1, Intel Core i5

The option `Use Rosetta for x86/amd64 emulation on Apple Silicon` resolves the issue for me: M1 2020, Ventura 13.6.4, Docker Desktop 4.27.1

**Comment by utkarz18** [comment]
I wasted my entire day to figure out what was wrong with the Seg fault and then after searching somehow came across this forum. Thanks for the work around guys and I can confirm that the issue persists!

Docker 4.27.1, MacOS 14.3 , Intel Core i5

**Comment by amosaikman** [comment]
Experienced problems with Remote Extension Host repeatedly crashing without obvious explanation. Also unexpected errors when trying to install PyTorch in a container. Disappeared when switched to gRPC and turned off Virtualisation Framework. Intel Mac Core i7 running macOS 14.3.1

**Comment by dschneiderch** [comment]
I also ran into this today. docker versions 4.27.2 and 4.28
switching to grpc and unchecking virtualization resolved it. i got here from https://github.com/microsoft/vscode/issues/194458

**Comment by AbhishekBhatiaQB** [comment]
@dossy Can't thank you enough for the solution!🙏 Confirming that switching to grpc and unchecking virtualization resolves it! ✅

**Comment by colbygk** [comment]
Can comfirm, seeing this on images derived from node:20 and node:18 on intel based macos. In our case, it manifests when attempting to use `tap` to run unit tests and it is not always immediate/nor in the same place each time.

**Comment by BrenoAccioly** [comment]
> https://docs.docker.com/desktop/release-notes/#4180

That worked for me! Many Thanks! I was having a similar issue.
Ventura 13.5.1, Intel Core i7, Docker 4.26.1

**Comment by BrenoAccioly** [comment]
> As an additional datapoint, I ran into what I imagine is this issue today with node v18.16.0 and running webpack 5.89.0, segfaulting on Docker 4.26.1 on Intel iMac 27" 2020 on macOS Ventura 13.3.1.
> 
> Switching from VirtioFS to gRPC FUSE and unchecking "Use Virtualization framework" and restarting Docker appears to have fixed the issue for me, as others have reported.

That worked for me! Many Thanks! I was having a similar issue.
Ventura 13.5.1, Intel Core i7, Docker 4.26.1

**Comment by panosru** [comment]
I tried to switch from `VirtioFS` to either `gRPC FUSE` or `osxfs (Legacy)` to disable the `Use Virtualization framework`, but neither worked because Docker was constantly crashing and wanted to reset.

**Comment by ddimitrioglo** [comment]
Switching to `gRPC FUSE` and unchecking `Use Virtualization framework` worked for me.

**Comment by SheepReaper** [comment]
The workaround is great and all, but the goal is to actually use the Virtualization Framework. Is there progress on identifying whether this is a VF, docker, or node problem? Individually all of these things work, just not together.

**Comment by ddimitrioglo** [comment]
@SheepReaper 
I'm using Docker w/ VF enabled to manage dozens of node services, so it's not that Docker (+VF) + Node.js combination.
I've got an error when I was trying to move from `babel` to `vite` or `parcel` builders.
Maybe that would give you a direction

**Comment by SheepReaper** [comment]
my issue manifested either when running yarn or running unit tests, jest or maybe karma. Idr, I haven't had the project open in a while (like since this issue was opened).

**Comment by TheRealDonaldDuck** [comment]
Confirming this issue on macOS Sonoma 14.4.1, Intel Core i7, and Docker v4.28.0, Docker Engine v25.0.3.

Using the gRPC without virtualization avoids the crashes but decreases performance in the devcontainer dramatically (many file accesses for compilation in C) so docker desktop is almost not usable anymore for me.

I wonder if there is any progress going on as this issue is pending for a quite long time.

**Comment by testacode** [comment]
> I tried to switch from `VirtioFS` to either `gRPC FUSE` or `osxfs (Legacy)` to disable the `Use Virtualization framework`, but neither worked because Docker was constantly crashing and wanted to reset.
> 
> macOS Sonoma v14.4 & Docker Desktop v4.28 (Engine: 25.0.3)

Same here, shit's unusable

**Comment by ralf57** [comment]
The issue is still present on macOS Sonoma v14.4.1 & Docker Desktop 4.29.0.
Disabling the Virtualization framework is not an option for me since performances decrease down to make it unusable.

**Comment by antoniofrignani** [comment]
I've switched to OrbStack since the issue popped up and solved the problem

**Comment by schwemmer** [comment]
I've also switched to OrbStack and have no regrets. Even on my Apple Silicon Macs that don't experience this issue with Docker Desktop, I have switched to OrbStack and it's just faster than Docker Desktop.

**Comment by pablo-psi** [comment]
> I've switched to OrbStack since the issue popped up and solved the problem
> 
> Edit: I've just realized it's been one year since I opened the issue and no one GAF about that, so anybody who will stumble upon this lately: do yourelf a favor and switch tool, this is not going to be fixed anytime soon.

Cheers @antoniofrignani, I didn't know about it! Seriously considering the switch now 🙌

**Comment by panosru** [comment]
I just switched! I previously switched to Rancher Desktop but eventually, I found Rancher lacking some features I liked in DD, so now I tried OrbStack and it seems to have all I need. Thanks!

**Comment by dgageot** [comment]
Hello everyone, so sorry for breaking your workflow! I'm currently trying to reproduce your issue, although there might be more than one.
In the meantime, if anyone of you is able to reproduce with a simple Dockerfile or a GitHub project that I cal build, that will help understand the root cause faster.

**Comment by eweidner** [comment]
@dgageot I found this to fail reliably on any system that showed the failure...

```
docker run -it node:18-slim bash;
npm install -g @angular/cli;
ng version;
```

**Comment by dgageot** [comment]
> @dgageot I found this to fail reliably on any system that showed the failure...
> 
> ```
> docker run -it node:18-slim bash;
> npm install -g @angular/cli;
> ng version;
> ```

Thanks a lot! I can't reproduce on Docker Desktop 4.29, on an Intel MacPro.
Can you share which version of DD you use and which machine/OS you have?

**Comment by SheepReaper** [comment]
> @dgageot I found this to fail reliably on any system that showed the failure...
> 
> ```
> docker run -it node:18-slim bash;
> npm install -g @angular/cli;
> ng version;
> ```

one liner: `docker run -it -e NG_CLI_ANALYTICS=false node:18-slim npx --yes --package @angular/cli -- ng version`

**Comment by dgageot** [comment]
Thanks @SheepReaper, that's handy! I've tried with this command too and same result: it works well. A colleague of mine has tried too and couldn't reproduce either.

Have you tried Docker Desktop 4.29? Can you share your version of macOS and type of Mac?

**Comment by sergioharo** [comment]
@dgageot I can reproduce on 4.29, iMac 2020 Intel i9 running Sonoma 14.4.1

You need to make sure you have `Use Virtualization Framework` enabled AND `VirtioFS` enabled. 

Running the test command above gets me

> docker run -it -e NG_CLI_ANALYTICS=false node:18-slim npx --yes --package @angular/cli -- ng version
Segmentation fault
npm notice 
npm notice New minor version of npm available! 10.5.0 -> 10.6.0
npm notice Changelog: https://github.com/npm/cli/releases/tag/v10.6.0
npm notice Run npm install -g npm@10.6.0 to update!
npm notice 

Using Virtualization Framework AND gRPC Fuse seems to work. 

As does NO virtualization framework AND osxfs 

Strangely after going back to VirtioFS, `node:18-slim` started working but `node:19-slim` still gets a segfault. Node 20+ seem to work.

**Comment by eweidner** [comment]
@dgageot I don't have a machine in a state to test this again right now, but as of May 2023, Nov 2023, and late Feb 2024 with all machines up to date on latest MacOS and Docker Desktop at that time I had 4 13" MBP (2020, Intel, i7 cpu) and 1 13" Macbook Air (2020, Intel, i7) all failed on this. I had 1 15" MBP (2018, Intel, i7) and 1 16" MBP (2019, Intel, i9) that did NOT have the problem with Mac Virtualization Framework on.

Turning off the Mac Virtualization Framework fixed all machines.

Also in my testing in November, node:16-slim did NOT have the problem. node:18-slim and node:20-slim DID have the problem. node:22-slim (not GA yet) did NOT have the problem.

Hope that helps.

**Comment by dgageot** [comment]
Keeping you posted: For now, we failed to reproduce on four different Intel Macs. Each time, we enabled `Virtualization Framework` enabled AND `VirtioFS`.
FWIW, I wonder how`VirtioFS` could have an impact, since both of the reproductions you shared don't use file sharing at all.

**Comment by dgageot** [comment]
OK, we've tried six Mac<=2018 and had no issue
But we found a 2020 MacBookPro with the issue!
I'll keep you posted

**Comment by antoniofrignani** [comment]
@dgageot thank for stepping in.

If that helps, you can have a look at the diagnostics i have uploaded when i first opened the issue.
Unfortunately I've switched to another tool the get the job done, so i can't help with the issue anymore.

**Comment by dgageot** [comment]
Here is an internal build which we think solves this issue. Could you give it a try?
https://desktop-stage.docker.com/mac/main/amd64/148395/Docker.dmg

**Comment by Amli** [comment]
A few days ago we upgraded our stack to nodejs 20 and suffered the same issue as described here (precisely a segfault when running webpack during image build with no mount) on a macbook pro 2020.

@dgageot I gave your internal build a try, switched back to use of the Virtualization Framework and used it yesterday for a few hours, everything works fine now 👏

I'll let others confirm that it works for them too.

Thanks for taking the time to reproduce and fix the issue !

**Comment by http-teapot** [comment]
I can confirm that it works on our end, too. Thank you for fixing this issue!

**Comment by dgageot** [comment]
FYI, the fix is planned for Docker Desktop v4.30.0

**Comment by pablo-psi** [comment]
> FYI, the fix is planned for Docker Desktop v4.30.0

Great, thank you for fixing the issue! 🙏

**Comment by chaizeg** [comment]
Closing this issue because a fix has been released in Docker Desktop `4.30.0` . See the [release notes](https://docs.docker.com/desktop/release-notes/) for more details.

**Comment by ciekawy** [comment]
Interestingly I am facing yarn SIGSEGV on Docker Desktom 4.33... Do I need to check specific versions/settings? I have the VirtioFS enabled
